src/gluonts/dataset/tabular.py

```diff
@@ -0,0 +1,269 @@
+        Name of the column that contains the timestamp information.
+    freq: str
+        Frequency of observations in the time series. Must be a valid pandas frequency.
+    feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features.
+    feat_dynamic_cat: List[str]
+        List of column names that contain dynamic categorical features.
+    feat_static_real: List[str]
+        List of column names that contain static real features.
+    feat_static_cat: List[str]
+        List of column names that contain static categorical features.
+    past_feat_dynamic_real: List[str]
+        List of column names that contain dynamic real features only for the history.
+    prediction_length: int
+        For target and past dynamic features last `prediction_length` elements are
+        removed when iterating over the data set. This becomes important when the
+        predictor is called.
+    """
+
+    dataframes: Union[List[pd.DataFrame], Dict[str, pd.DataFrame]]
+    target: Union[str, List[str]]
+    timestamp: str
+    freq: str
+    feat_dynamic_real: List[str] = field(default_factory=list)
+    feat_dynamic_cat: List[str] = field(default_factory=list)
+    feat_static_real: List[str] = field(default_factory=list)
+    feat_static_cat: List[str] = field(default_factory=list)
+    past_feat_dynamic_real: List[str] = field(default_factory=list)
+    prediction_length: int = 0
+
+    def __post_init__(self) -> None:
+        self.one_dim_target = not isinstance(self.target, list)
+        self.process = ProcessDataEntry(
+            self.freq, one_dim_target=self.one_dim_target
+        )
+        if isinstance(self.dataframes, dict):
+            self._dataframes = list(self.dataframes.items())
+        else:  # case list
+            self._dataframes = [("", df) for df in self.dataframes]
+
+    def _dataentry(self, item_id: str, df: pd.DataFrame) -> DataEntry:
+        check_timestamps(df.loc[:, self.timestamp].to_list(), freq=self.freq)
+        dataentry = dataframeTS_to_dataentry(
+            data=df,
+            target=self.target,
+            timestamp=self.timestamp,
+            feat_dynamic_real=self.feat_dynamic_real,
+            feat_dynamic_cat=self.feat_dynamic_cat,
+            feat_static_real=self.feat_static_real,
+            feat_static_cat=self.feat_static_cat,
+            past_feat_dynamic_real=self.past_feat_dynamic_real,
+        )
+        if item_id:
+            dataentry["item_id"] = item_id
+        return dataentry
+
+    def __iter__(self) -> Iterator[DataEntry]:
+        for item_id, df in self._dataframes:
+            dataentry = self.process(self._dataentry(item_id, df))
+            if self.prediction_length:
+                dataentry = prepare_prediction_data(
+                    dataentry, self.prediction_length
+                )
+            yield dataentry
+
+    def __len__(self) -> int:
+        return len(self._dataframes)
+
+
+def LongDataFrameDataset(
```

Comment:
If this is a function, why does it use camel-case?

Reply:
We could have an alternative constructor for `DataFramesDataset` instead. What's the advantage to make this standalone?